[Localization] localized status-effect.ts and translate.

EnochG1 opened the pull request:
## What are the changes?
Add the option to localize message text of status effects. (at status-effects.ts)

## Why am I doing these changes?
All text should be localizable, status effect message is important to play the game.

## What did change?
 - Change the code for localize
 - Added config and new locale files > status-effect.ts
 - Added localization key in each language folders.
 - Translated of them to Korean.

### Screenshots/Videos
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/15914ef5-b789-4967-8319-5b3e90d3dc51)
English

![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/16e51655-aba4-4bb8-878f-60e9bdda50c7)
Korean


## How to test the changes?
 - Using test code : Run ```src/test/lokalisation/status-effect.test.ts```
 - Manually : modify overrides.ts to test status-effect using MOVESET, for example.
```typescript
export const WEATHER_OVERRIDE: WeatherType = WeatherType.SNOW; // To increase the probability of Freeze if need
export const STARTING_BIOME_OVERRIDE: Biome = Biome.ICE_CAVE; // To increase the probability of Freeze if need

export const MOVESET_OVERRIDE: Array<Moves> = [ Moves.POISON_GAS, Moves.TOXIC, Moves.NUZZLE, Moves.SPORE ];
// export const MOVESET_OVERRIDE: Array<Moves> = [ Moves.SIZZLY_SLIDE, Moves.TOXIC_SPIKES, Moves.POWDER_SNOW ];
```


## Checklist
- [ ] There is no overlap with another PR?
- [ ] The PR is self-contained and cannot be split into smaller PRs?
- [ ] Have I provided a clear explanation of the changes?
- [ ] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

40chyan commented on src/locales/zh_CN/status-effect.ts:
```diff
@@ -0,0 +1,67 @@
+    overlap: "{{pokemonNameWithAffix}} is\nalready poisoned!",
+    heal: "{{pokemonNameWithAffix}} was\ncured of its poison!"
+  },
+  paralysis: {
+    name: "Paralysis",
+    description: "paralysis",
+    obtain: "{{pokemonNameWithAffix}} was paralyzed,\nIt may be unable to move!",
+    obtainSource: "{{pokemonNameWithAffix}} was paralyzed by {{sourceText}},\nIt may be unable to move!",
+    activation: "{{pokemonNameWithAffix}} is paralyzed!\nIt can't move!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready paralyzed!",
+    heal: "{{pokemonNameWithAffix}} was\nhealed of paralysis!"
+  },
+  sleep: {
+    name: "Sleep",
+    description: "sleep",
+    obtain: "{{pokemonNameWithAffix}}\nfell asleep!",
+    obtainSource: "{{pokemonNameWithAffix}}\nfell asleep from {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is fast asleep.",
+    overlap: "{{pokemonNameWithAffix}} is\nalready asleep!",
+    heal: "{{pokemonNameWithAffix}} woke up!"
+  },
+  freeze: {
+    name: "Freeze",
+    description: "freezing",
+    obtain: "{{pokemonNameWithAffix}}\nwas frozen solid!",
+    obtainSource: "{{pokemonNameWithAffix}}\nwas frozen solid by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is\nfrozen solid!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready frozen!",
+    heal: "{{pokemonNameWithAffix}} was\ndefrosted!"
+  },
+  burn: {
+    name: "Burn",
+    description: "burn",
+    obtain: "{{pokemonNameWithAffix}}\nwas burned!",
+    obtainSource: "{{pokemonNameWithAffix}}\nwas burned by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is hurt\nby its burn!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready burned!",
+    heal: "{{pokemonNameWithAffix}} was\nhealed of its burn!"
+  },
+} as const;
```
```suggestion
export const statusEffect: StatusEffectTranslationEntries = {
  none: {
    name: "无",
    description: "",
    obtain: "",
    obtainSource: "",
    activation: "",
    overlap: "",
    heal: ""
  },
  poison: {
    name: "中毒",
    description: "中毒",
    obtain: "{{pokemonNameWithAffix}}中毒了！",
    obtainSource: "{{pokemonNameWithAffix}}因{{sourceText}}中毒了！",
    activation: "{{pokemonNameWithAffix}}受到了毒的伤害！",
    overlap: "{{pokemonNameWithAffix}}已经中毒了！",
    heal: "{{pokemonNameWithAffix}}中的毒彻底清除了！"
  },
  toxic: {
    name: "剧毒",
    description: "中毒",
    obtain: "{{pokemonNameWithAffix}}中了剧毒！",
    obtainSource: "{{pokemonNameWithAffix}}因{{sourceText}}中了剧毒！",
    activation: "{{pokemonNameWithAffix}}受到了毒的伤害！",
    overlap: "{{pokemonNameWithAffix}}已经中毒了！",
    heal: "{{pokemonNameWithAffix}}中的毒彻底清除了！"
  },
  paralysis: {
    name: "麻痹",
    description: "麻痹",
    obtain: "{{pokemonNameWithAffix}}麻痹了，很难使出招式！",
    obtainSource: "{{pokemonNameWithAffix}}被{{sourceText}}麻痹了，很难使出招式！",
    activation: "{{pokemonNameWithAffix}}因身体麻痹而无法行动！",
    overlap: "{{pokemonNameWithAffix}}已经麻痹了！",
    heal: "{{pokemonNameWithAffix}}的麻痹治愈了！"
  },
  sleep: {
    name: "睡眠",
    description: "睡眠",
    obtain: "{{pokemonNameWithAffix}}睡着了！",
    obtainSource: "{{pokemonNameWithAffix}}因{{sourceText}}睡着了！",
    activation: "{{pokemonNameWithAffix}}正在呼呼大睡。",
    overlap: "{{pokemonNameWithAffix}}已经睡着了！",
    heal: "{{pokemonNameWithAffix}}醒了！"
  },
  freeze: {
    name: "冰冻",
    description: "冰冻",
    obtain: "{{pokemonNameWithAffix}}冻住了！",
    obtainSource: "{{pokemonNameWithAffix}}因{{sourceText}}冻住了！",
    activation: "{{pokemonNameWithAffix}}因冻住了而无法行动！",
    overlap: "{{pokemonNameWithAffix}}已经冻住了！",
    heal: "{{pokemonNameWithAffix}}治愈了冰冻状态！"
  },
  burn: {
    name: "灼伤",
    description: "灼伤",
    obtain: "{{pokemonNameWithAffix}}被灼伤了！",
    obtainSource: "{{pokemonNameWithAffix}}因{{sourceText}}被灼伤了！",
    activation: "{{pokemonNameWithAffix}}受到了灼伤的伤害！",
    overlap: "{{pokemonNameWithAffix}}已经被灼伤了！",
    heal: "{{pokemonNameWithAffix}}的灼伤治愈了！"
  },
} as const;
```

Adri1 commented on src/locales/fr/status-effect.ts:
```diff
@@ -0,0 +1,67 @@
+    obtainSource: "{{pokemonNameWithAffix}}\nwas badly poisoned by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is hurt\nby poison!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready poisoned!",
+    heal: "{{pokemonNameWithAffix}} was\ncured of its poison!"
+  },
+  paralysis: {
+    name: "Paralysis",
+    description: "paralysis",
+    obtain: "{{pokemonNameWithAffix}} was paralyzed,\nIt may be unable to move!",
+    obtainSource: "{{pokemonNameWithAffix}} was paralyzed by {{sourceText}},\nIt may be unable to move!",
+    activation: "{{pokemonNameWithAffix}} is paralyzed!\nIt can't move!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready paralyzed!",
+    heal: "{{pokemonNameWithAffix}} was\nhealed of paralysis!"
+  },
+  sleep: {
+    name: "Sleep",
+    description: "sleep",
+    obtain: "{{pokemonNameWithAffix}}\nfell asleep!",
+    obtainSource: "{{pokemonNameWithAffix}}\nfell asleep from {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is fast asleep.",
+    overlap: "{{pokemonNameWithAffix}} is\nalready asleep!",
+    heal: "{{pokemonNameWithAffix}} woke up!"
+  },
+  freeze: {
+    name: "Freeze",
+    description: "freezing",
+    obtain: "{{pokemonNameWithAffix}}\nwas frozen solid!",
+    obtainSource: "{{pokemonNameWithAffix}}\nwas frozen solid by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is\nfrozen solid!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready frozen!",
+    heal: "{{pokemonNameWithAffix}} was\ndefrosted!"
+  },
+  burn: {
+    name: "Burn",
+    description: "burn",
+    obtain: "{{pokemonNameWithAffix}}\nwas burned!",
+    obtainSource: "{{pokemonNameWithAffix}}\nwas burned by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is hurt\nby its burn!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready burned!",
+    heal: "{{pokemonNameWithAffix}} was\nhealed of its burn!"
```
```suggestion
    name: "Aucun",
    description: "",
    obtain: "",
    obtainSource: "",
    activation: "",
    overlap: "",
    heal: ""
  },
  poison: {
    name: "Empoisonnement",
    description: "empoisonnement",
    obtain: "{{pokemonNameWithAffix}}\nest empoisonné !",
    obtainSource: "{{pokemonNameWithAffix}}\nest empoisonné par {{sourceText}} !",
    activation: "{{pokemonNameWithAffix}}\nsouffre du poison !",
    overlap: "{{pokemonNameWithAffix}}\nest déjà empoisonné.",
    heal: "{{pokemonNameWithAffix}}\nn’est plus empoisonné !"
  },
  toxic: {
    name: "Empoisonnement grave",
    description: "empoisonnement grave",
    obtain: "{{pokemonNameWithAffix}}\nest gravement empoisonné !",
    obtainSource: "{{pokemonNameWithAffix}}\nest gravement empoisonné par {{sourceText}} !",
    activation: "{{pokemonNameWithAffix}}\nsouffre du poison !",
    overlap: "{{pokemonNameWithAffix}}\nest déjà empoisonné.",
    heal: "{{pokemonNameWithAffix}}\nn’est plus empoisonné !"
  },
  paralysis: {
    name: "Paralysie",
    description: "paralysie",
    obtain: "{{pokemonNameWithAffix}} est paralysé !\nIl aura du mal à attaquer !",
    obtainSource: "{{pokemonNameWithAffix}} est paralysé\npar {{sourceText}} ! Il aura du mal à attaquer !",
    activation: "{{pokemonNameWithAffix}} est paralysé !\nIl n’a pas pu attaquer !",
    overlap: "{{pokemonNameWithAffix}}\nest déjà paralysé.",
    heal: "{{pokemonNameWithAffix}}\nn’est plus paralysé !"
  },
  sleep: {
    name: "Sommeil",
    description: "sommeil",
    obtain: "{{pokemonNameWithAffix}}\ns’est endormi !",
    obtainSource: "{{pokemonNameWithAffix}}\nest endormi par {{sourceText}} !",
    activation: "{{pokemonNameWithAffix}}\ndort profondément.",
    overlap: "{{pokemonNameWithAffix}}\ndort déjà.",
    heal: "{{pokemonNameWithAffix}}\se réveille !"
  },
  freeze: {
    name: "Gelé",
    description: "gelé",
    obtain: "{{pokemonNameWithAffix}}\nest gelé !",
    obtainSource: "{{pokemonNameWithAffix}}\nest gelé par {{sourceText}} !",
    activation: "{{pokemonNameWithAffix}}est gelé !\nIl ne peut plus attaquer !",
    overlap: "{{pokemonNameWithAffix}}\nest déjà gelé.",
    heal: "{{pokemonNameWithAffix}}\nn’est plus gelé !"
  },
  burn: {
    name: "Brulure",
    description: "brulure",
    obtain: "{{pokemonNameWithAffix}}\nest brulé !",
    obtainSource: "{{pokemonNameWithAffix}}\nest brulé par {{sourceText}} !",
    activation: "{{pokemonNameWithAffix}}\nsouffre de sa brulure !",
    overlap: "{{pokemonNameWithAffix}}\nest déjà brulé.",
    heal: "{{pokemonNameWithAffix}}\nn’est plus brulé !"
```

Adri1 reviewed (commented):
French translations

CodeTappert commented on src/locales/de/status-effect.ts:
```diff
@@ -0,0 +1,67 @@
+    obtainSource: "{{pokemonNameWithAffix}}\nwas badly poisoned by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is hurt\nby poison!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready poisoned!",
+    heal: "{{pokemonNameWithAffix}} was\ncured of its poison!"
+  },
+  paralysis: {
+    name: "Paralysis",
+    description: "paralysis",
+    obtain: "{{pokemonNameWithAffix}} was paralyzed,\nIt may be unable to move!",
+    obtainSource: "{{pokemonNameWithAffix}} was paralyzed by {{sourceText}},\nIt may be unable to move!",
+    activation: "{{pokemonNameWithAffix}} is paralyzed!\nIt can't move!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready paralyzed!",
+    heal: "{{pokemonNameWithAffix}} was\nhealed of paralysis!"
+  },
+  sleep: {
+    name: "Sleep",
+    description: "sleep",
+    obtain: "{{pokemonNameWithAffix}}\nfell asleep!",
+    obtainSource: "{{pokemonNameWithAffix}}\nfell asleep from {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is fast asleep.",
+    overlap: "{{pokemonNameWithAffix}} is\nalready asleep!",
+    heal: "{{pokemonNameWithAffix}} woke up!"
+  },
+  freeze: {
+    name: "Freeze",
+    description: "freezing",
+    obtain: "{{pokemonNameWithAffix}}\nwas frozen solid!",
+    obtainSource: "{{pokemonNameWithAffix}}\nwas frozen solid by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is\nfrozen solid!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready frozen!",
+    heal: "{{pokemonNameWithAffix}} was\ndefrosted!"
+  },
+  burn: {
+    name: "Burn",
+    description: "burn",
+    obtain: "{{pokemonNameWithAffix}}\nwas burned!",
+    obtainSource: "{{pokemonNameWithAffix}}\nwas burned by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is hurt\nby its burn!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready burned!",
+    heal: "{{pokemonNameWithAffix}} was\nhealed of its burn!"
```
```suggestion
    name: "Gift",
    description: "Vergiftungen",
    obtain: "{{pokemonNameWithAffix}} wurde vergiftet!",
    obtainSource: "{{pokemonNameWithAffix}} wurde durch {{sourceText}} vergiftet!",
    activation: "{{pokemonNameWithAffix}} wird durch Gift verletzt!",
    overlap: "{{pokemonNameWithAffix}} ist bereits vergiftet!",
    heal: "Die Vergiftung von {{pokemonNameWithAffix}} wurde geheilt!"
  },
  toxic: {
    name: "Gift",
    description: "Vergiftungen",
    obtain: "{{pokemonNameWithAffix}} wurde schwer vergiftet!",
    obtainSource: "{{pokemonNameWithAffix}} wurde durch  {{sourceText}} schwer vergiftet!",
    activation: "{{pokemonNameWithAffix}} wird durch Gift verletzt!",
    overlap: "{{pokemonNameWithAffix}} ist bereits vergiftet!",
    heal: "Die Vergiftung von {{pokemonNameWithAffix}} wurde geheilt!"
  },
  paralysis: {
    name: "Paralyse",
    description: "Paralyse",
    obtain: "{{pokemonNameWithAffix}} wurde paralysiert!\nEs kann eventuell nicht handeln!",
    obtainSource: "{{pokemonNameWithAffix}} wurde durch {{sourceText}} paralysiert,\nEs kann eventuell nicht handeln!",
    activation: "{{pokemonNameWithAffix}}ist paralysiert!\nEs kann nicht angreifen!",
    overlap: "{{pokemonNameWithAffix}} ist bereits paralysiert!",
    heal: "Die Paralyse von {{pokemonNameWithAffix}} wurde aufgehoben!"
  },
  sleep: {
    name: "Schlaf",
    description: "Einschlafen",
    obtain: "{{pokemonNameWithAffix}} ist eingeschlafen!",
    obtainSource: "{{pokemonNameWithAffix}}ist durch {{sourceText}} eingeschlafen!",
    activation: "{{pokemonNameWithAffix}} schläft tief und fest!",
    overlap: "{{pokemonNameWithAffix}} schläft bereits!",
    heal: "{{pokemonNameWithAffix}} ist aufgewacht!"
  },
  freeze: {
    name: "Gefroren",
    description: "Einfrieren",
    obtain: "{{pokemonNameWithAffix}} erstarrt zu Eis!",
    obtainSource: "{{pokemonNameWithAffix}} erstarrt durch  {{sourceText}} zu Eis!",
    activation: "{{pokemonNameWithAffix}} ist eingefroren und kann nicht handeln!",
    overlap: "{{pokemonNameWithAffix}} ist bereits eingefroren!",
    heal: "{{pokemonNameWithAffix}} wurde aufgetaut!"
  },
  burn: {
    name: "Verbrennung ",
    description: "Verbrennungen",
    obtain: "{{pokemonNameWithAffix}} erleidet Verbrennungen!",
    obtainSource: "{{pokemonNameWithAffix}} erleidet durch {{sourceText}} Verbrennungen!",
    activation: "Die Verbrennungen schaden {{pokemonNameWithAffix}}!",
    overlap: "{{pokemonNameWithAffix}} leidet bereits unter
Verbrennungen!",
    heal: "Die Verbrennungen von {{pokemonNameWithAffix}}  wurden geheilt!"
```

EnochG1 commented on src/locales/de/status-effect.ts:
```diff
@@ -0,0 +1,68 @@
+    obtain: "{{pokemonNameWithAffix}} erleidet Verbrennungen!",
+    obtainSource: "{{pokemonNameWithAffix}} erleidet durch {{sourceText}} Verbrennungen!",
+    activation: "Die Verbrennungen schaden {{pokemonNameWithAffix}}!",
+    overlap: "{{pokemonNameWithAffix}} leidet bereits unter
+Verbrennungen!",
```
```suggestion
    overlap: "{{pokemonNameWithAffix}} leidet bereits unter Verbrennungen!",
```

OrangeRed commented on src/test/lokalisation/status-effect.test.ts:
```diff
@@ -0,0 +1,185 @@
+    statusEffectUnits.push({ stat: StatusEffect.BURN, key: "burn" });
+  });
+
+  it("Test getStatusEffectMessages in English", async () => {
+    i18next.changeLanguage("en");
+    statusEffectUnits.forEach(unit => {
+      testStatusEffectObtainText(unit.stat, enStatusEffect[unit.key].obtain);
+      testStatusEffectObtainTextWithSource(unit.stat, enStatusEffect[unit.key].obtainSource);
+      testStatusEffectActivationText(unit.stat, enStatusEffect[unit.key].activation);
+      testStatusEffectOverlapText(unit.stat, enStatusEffect[unit.key].overlap);
+      testStatusEffectHealText(unit.stat, enStatusEffect[unit.key].heal);
+      testStatusEffectDescriptor(unit.stat, enStatusEffect[unit.key].description);
+    });
+  });
```
I don't think these tests are necessary since you're not really testing anything that we created

EnochG1 replied:
Then any suggestion to improve test code (or way to get review) for making this PR approve?

EnochG1 replied:
My test code is for checking this.
1. Is i18n key matching right?
2. Are translators wrote json parameter key right?
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/6ffe8ca4-303b-44a3-b55b-599182c8c5a3)

flx-sta commented on src/test/lokalisation/status-effect.test.ts:
```diff
@@ -0,0 +1,185 @@
+    statusEffectUnits.push({ stat: StatusEffect.BURN, key: "burn" });
+  });
+
+  it("Test getStatusEffectMessages in English", async () => {
+    i18next.changeLanguage("en");
+    statusEffectUnits.forEach(unit => {
+      testStatusEffectObtainText(unit.stat, enStatusEffect[unit.key].obtain);
+      testStatusEffectObtainTextWithSource(unit.stat, enStatusEffect[unit.key].obtainSource);
+      testStatusEffectActivationText(unit.stat, enStatusEffect[unit.key].activation);
+      testStatusEffectOverlapText(unit.stat, enStatusEffect[unit.key].overlap);
+      testStatusEffectHealText(unit.stat, enStatusEffect[unit.key].heal);
+      testStatusEffectDescriptor(unit.stat, enStatusEffect[unit.key].description);
+    });
+  });
```
```suggestion
  it("Displays posion text", async () => {
   // i18next.changeLanguage("en"); DONT set up a language. make it print only the translation key!
      const text = getStatusEffectObtainText(unit.stat, 'some pokemon name i guess;');
      
      expect(text).toBe('stat:poison.text')
  });
```

EnochG1 replied:
I added test code for check key string as you advised. Thank you for your help.
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/23d1d956-8679-485c-87cd-54d3c0239a2a)
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/25439062-0b2b-4e34-991b-8b9e34079546)

EnochG1 commented:
@Tempo-anon ,
I want to get your review, espacially about test code. (Because you're main reviewer of localization, normally.)
I wanted to try not to make mistake like https://github.com/pagefaultgames/pokerogue/pull/2410 .
Also, I thought once I make test code, it can be prevented after I got translation from others without manual test of all cases.
(Checking if i18n key matches, json parameter key is right, and so on...)
Any advice for making easy to review and be approved?

josericardo-fo commented on src/locales/pt_BR/status-effect.ts:
```diff
@@ -0,0 +1,67 @@
+    obtainSource: "{{pokemonNameWithAffix}}\nwas badly poisoned by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is hurt\nby poison!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready poisoned!",
+    heal: "{{pokemonNameWithAffix}} was\ncured of its poison!"
+  },
+  paralysis: {
+    name: "Paralysis",
+    description: "paralysis",
+    obtain: "{{pokemonNameWithAffix}} was paralyzed,\nIt may be unable to move!",
+    obtainSource: "{{pokemonNameWithAffix}} was paralyzed by {{sourceText}},\nIt may be unable to move!",
+    activation: "{{pokemonNameWithAffix}} is paralyzed!\nIt can't move!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready paralyzed!",
+    heal: "{{pokemonNameWithAffix}} was\nhealed of paralysis!"
+  },
+  sleep: {
+    name: "Sleep",
+    description: "sleep",
+    obtain: "{{pokemonNameWithAffix}}\nfell asleep!",
+    obtainSource: "{{pokemonNameWithAffix}}\nfell asleep from {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is fast asleep.",
+    overlap: "{{pokemonNameWithAffix}} is\nalready asleep!",
+    heal: "{{pokemonNameWithAffix}} woke up!"
+  },
+  freeze: {
+    name: "Freeze",
+    description: "freezing",
+    obtain: "{{pokemonNameWithAffix}}\nwas frozen solid!",
+    obtainSource: "{{pokemonNameWithAffix}}\nwas frozen solid by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is\nfrozen solid!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready frozen!",
+    heal: "{{pokemonNameWithAffix}} was\ndefrosted!"
+  },
+  burn: {
+    name: "Burn",
+    description: "burn",
+    obtain: "{{pokemonNameWithAffix}}\nwas burned!",
+    obtainSource: "{{pokemonNameWithAffix}}\nwas burned by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is hurt\nby its burn!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready burned!",
+    heal: "{{pokemonNameWithAffix}} was\nhealed of its burn!"
```
```suggestion
    name: "Nenhum",
    description: "",
    obtain: "",
    obtainSource: "",
    activation: "",
    overlap: "",
    heal: ""
  },
  poison: {
    name: "Envenenamento",
    description: "envenenamento",
    obtain: "{{pokemonNameWithAffix}}\nfoi envenenado!",
    obtainSource: "{{pokemonNameWithAffix}}\nfoi envenenado por {{sourceText}}!",
    activation: "{{pokemonNameWithAffix}} foi ferido\ncom o veneno!",
    overlap: "{{pokemonNameWithAffix}} já\nestá envenenado!",
    heal: "{{pokemonNameWithAffix}} se\ncurou do envenenamento!"
  },
  toxic: {
    name: "Toxic",
    description: "envenenamento",
    obtain: "{{pokemonNameWithAffix}}\nfoi seriamente envenenado!",
    obtainSource: "{{pokemonNameWithAffix}} foi seriamente\nenvenenado por {{sourceText}}!",
    activation: "{{pokemonNameWithAffix}} foi ferido\ncom o veneno!",
    overlap: "{{pokemonNameWithAffix}} já\nestá envenenado!",
    heal: "{{pokemonNameWithAffix}} se\ncurou do envenenamento!"
  },
  paralysis: {
    name: "Paralisia",
    description: "paralisia",
    obtain: "{{pokemonNameWithAffix}} foi paralisado,\nTalvez ele não consiga se mover!",
    obtainSource: "{{pokemonNameWithAffix}} foi paralisado por {{sourceText}},\nTalvez ele não consiga se mover!",
    activation: "{{pokemonNameWithAffix}} está paralisado!\nEle não consegue se mover!",
    overlap: "{{pokemonNameWithAffix}} já\nestá paralisado!",
    heal: "{{pokemonNameWithAffix}} foi\ncurado da paralisia!"
  },
  sleep: {
    name: "Dormindo",
    description: "dormindo",
    obtain: "{{pokemonNameWithAffix}}\nadormeceu!",
    obtainSource: "{{pokemonNameWithAffix}}\ndormiu devido a {{sourceText}}!",
    activation: "{{pokemonNameWithAffix}} está dormindo profundamente.",
    overlap: "{{pokemonNameWithAffix}} já\nestá dormindo!",
    heal: "{{pokemonNameWithAffix}} acordou!"
  },
  freeze: {
    name: "Congelamento",
    description: "congelando",
    obtain: "{{pokemonNameWithAffix}}\nfoi congelado!",
    obtainSource: "{{pokemonNameWithAffix}}\nfoi congelado por {{sourceText}}!",
    activation: "{{pokemonNameWithAffix}} está\ncongelado!",
    overlap: "{{pokemonNameWithAffix}} já\nestá congelado!",
    heal: "{{pokemonNameWithAffix}} foi\ndescongelado!"
  },
  burn: {
    name: "Queimadura",
    description: "queimadura",
    obtain: "{{pokemonNameWithAffix}}\nfoi queimado!",
    obtainSource: "{{pokemonNameWithAffix}}\nfoi queimado por {{sourceText}}!",
    activation: "{{pokemonNameWithAffix}} foi ferido\npor sua queimadura!",
    overlap: "{{pokemonNameWithAffix}} já\nestá queimado!",
    heal: "{{pokemonNameWithAffix}} foi\ncurado de sua queimadura!"
```

EnochG1 commented:
@flx-sta Could you re-review this PR?
I removed multi-language test codes except english (for check if message method runs right)

EnochG1 commented:
@flx-sta
Could you re-review this PR?
I removed language, and changed test code to validate if method call and parameter execute right.
(Used toHaveBeenCalledWith instead of mock function)

GoldTra commented on src/locales/es/status-effect.ts:
```diff
@@ -0,0 +1,67 @@
+    obtainSource: "{{pokemonNameWithAffix}}\nwas badly poisoned by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is hurt\nby poison!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready poisoned!",
+    heal: "{{pokemonNameWithAffix}} was\ncured of its poison!"
+  },
+  paralysis: {
+    name: "Paralysis",
+    description: "paralysis",
+    obtain: "{{pokemonNameWithAffix}} was paralyzed,\nIt may be unable to move!",
+    obtainSource: "{{pokemonNameWithAffix}} was paralyzed by {{sourceText}},\nIt may be unable to move!",
+    activation: "{{pokemonNameWithAffix}} is paralyzed!\nIt can't move!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready paralyzed!",
+    heal: "{{pokemonNameWithAffix}} was\nhealed of paralysis!"
+  },
+  sleep: {
+    name: "Sleep",
+    description: "sleep",
+    obtain: "{{pokemonNameWithAffix}}\nfell asleep!",
+    obtainSource: "{{pokemonNameWithAffix}}\nfell asleep from {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is fast asleep.",
+    overlap: "{{pokemonNameWithAffix}} is\nalready asleep!",
+    heal: "{{pokemonNameWithAffix}} woke up!"
+  },
+  freeze: {
+    name: "Freeze",
+    description: "freezing",
+    obtain: "{{pokemonNameWithAffix}}\nwas frozen solid!",
+    obtainSource: "{{pokemonNameWithAffix}}\nwas frozen solid by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is\nfrozen solid!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready frozen!",
+    heal: "{{pokemonNameWithAffix}} was\ndefrosted!"
+  },
+  burn: {
+    name: "Burn",
+    description: "burn",
+    obtain: "{{pokemonNameWithAffix}}\nwas burned!",
+    obtainSource: "{{pokemonNameWithAffix}}\nwas burned by {{sourceText}}!",
+    activation: "{{pokemonNameWithAffix}} is hurt\nby its burn!",
+    overlap: "{{pokemonNameWithAffix}} is\nalready burned!",
+    heal: "{{pokemonNameWithAffix}} was\nhealed of its burn!"
```
```suggestion
  none: {
    name: "Ninguno",
    description: "",
    obtain: "",
    obtainSource: "",
    activation: "",
    overlap: "",
    heal: ""
  },
  poison: {
    name: "Envenenado",
    description: "envenenado ",
    obtain: "¡{{pokemonNameWithAffix}}\nha sido envenenado!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido envenenado por {{sourceText}}!",
    activation: "¡{{pokemonNameWithAffix}} se hizo daño\npor el veneno!",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá envenenado!",
    heal: "¡{{pokemonNameWithAffix}} se\ncuró del envenenamiento!"
  },
  toxic: {
    name: "Envenenado grave",
    description: "envenenado grave",
    obtain: "¡{{pokemonNameWithAffix}}\nha sido gravemente envenenado!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido gravemente envenenado por {{sourceText}}!",
    activation: "¡{{pokemonNameWithAffix}} se hizo daño\npor el veneno!",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá envenenado!",
    heal: "¡{{pokemonNameWithAffix}} se\ncuró del envenenamiento!"
  },
  paralysis: {
    name: "Paralizado",
    description: "paralizado",
    obtain: "¡{{pokemonNameWithAffix}} sufre parálisis!\nQuizás no se pueda mover.",
    obtainSource: "¡{{pokemonNameWithAffix}} sufre parálisis por {{sourceText}}!\nQuizás no se pueda mover.",
    activation: "¡{{pokemonNameWithAffix}} está paralizado!\n¡No se puede mover!",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá paralizado!",
    heal: "¡{{pokemonNameWithAffix}} se\ncuró de la parálisis !"
  },
  sleep: {
    name: "Dormido",
    description: "dormido",
    obtain: "¡{{pokemonNameWithAffix}}\nse durmió!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nse durmió por {{sourceText}}!",
    activation: "¡{{pokemonNameWithAffix}} está/ndormido como un tronco.",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá dormido!",
    heal: "¡{{pokemonNameWithAffix}} se despertó!"
  },
  freeze: {
    name: "Congelado",
    description: "congelado",
    obtain: "¡{{pokemonNameWithAffix}}\nha sido congelado!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido congelado por {{sourceText}}!",
    activation: "¡{{pokemonNameWithAffix}} está\ncongelado!",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá congelado!",
    heal: "¡{{pokemonNameWithAffix}} se\ndescongeló!"
  },
  burn: {
    name: "Quemado",
    description: "quemado",
    obtain: "¡{{pokemonNameWithAffix}}\nse ha quemado!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nse ha quemado por {{sourceText}}!",
    activation: "¡{{pokemonNameWithAffix}} se hizo daño\npor el quemado!",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá quemado!",
    heal: "¡{{pokemonNameWithAffix}} se\ncuró del quemado!"
```

GoldTra reviewed (changes requested):
Spanish translations

Adri1 commented on src/locales/en/status-effect.ts:
```diff
@@ -0,0 +1,67 @@
+    name: "Paralysis",
+    description: "paralysis",
+    obtain: "{{pokemonNameWithAffix}} was paralyzed,\nIt may be unable to move!",
+    obtainSource: "{{pokemonNameWithAffix}} was paralyzed by {{sourceText}},\nIt may be unable to move!",
```
```suggestion
    obtainSource: "{{pokemonNameWithAffix}} was paralyzed by {{sourceText}}!\nIt may be unable to move!",
```

Adri1 reviewed (commented):
Added a "!" instead of a coma (see official text below)

![image](https://github.com/pagefaultgames/pokerogue/assets/2070109/878e2707-34e0-4b4c-887f-b7fcf974e041)

flx-sta commented on src/test/lokalisation/status-effect.test.ts:
```diff
@@ -0,0 +1,92 @@
+function testStatusEffectOverlapText(stat: StatusEffect, expectKey: ParseKeys) {
+  const i18nextSpy = vi.spyOn(i18next, "t");
+  const message = getStatusEffectOverlapText(stat, testPokemonNameWithoutBinding);
+  console.log(`expectKey ${expectKey}, message ${message}`);
```
no console logs in final product pls

flx-sta commented on src/test/lokalisation/status-effect.test.ts:
```diff
@@ -0,0 +1,92 @@
+function testStatusEffectHealText(stat: StatusEffect, expectKey: ParseKeys) {
+  const i18nextSpy = vi.spyOn(i18next, "t");
+  const message = getStatusEffectHealText(stat, testPokemonNameWithoutBinding);
+  console.log(`expectKey ${expectKey}, message ${message}`);
```
no console logs in final product pls

flx-sta commented on src/test/lokalisation/status-effect.test.ts:
```diff
@@ -0,0 +1,92 @@
+function testStatusEffectDescriptor(stat: StatusEffect, expectKey: ParseKeys) {
+  const i18nextSpy = vi.spyOn(i18next, "t");
+  const message = getStatusEffectDescriptor(stat);
+  console.log(`expectKey ${expectKey}, message ${message}`);
```
no console logs in final product pls

flx-sta commented on src/test/lokalisation/status-effect.test.ts:
```diff
@@ -0,0 +1,92 @@
+    statusEffectUnits.forEach(unit => {
+      const text = getStatusEffectMessageKey(unit.stat);
+      const toBe = `statusEffect:${unit.key}`;
+      console.log(`message ${text}, expected ${toBe}`);
```
no console logs in final product pls

flx-sta commented on src/data/status-effect.ts:
```diff
@@ -31,94 +32,52 @@ export class Status {
-
-  return "";
+  const i18nKey = `${getStatusEffectMessageKey(statusEffect)}.obtainSource`as ParseKeys;
+  return !sourceText ? i18next.t(i18nKey, { pokemonNameWithAffix: pokemonNameWithAffix }) : i18next.t(i18nKey, { pokemonNameWithAffix: pokemonNameWithAffix, sourceText: sourceText });
```
Checking `!sourceText` here again seems redundant. Above this check was already done together with a `return` inside the if-clause

flx-sta commented:
@EnochG1 Here is how I would have written the tests.

<details>
  <summary>status-effect.test.ts</summary>

```ts
import {
  Status,
  StatusEffect,
  getStatusEffectActivationText,
  getStatusEffectDescriptor,
  getStatusEffectHealText,
  getStatusEffectObtainText,
  getStatusEffectOverlapText,
} from "#app/data/status-effect";
import i18next, { ParseKeys } from "i18next";
import { afterEach } from "node:test";
import { beforeAll, describe, expect, it, vi } from "vitest";

const tMock = (key: ParseKeys) => key;

describe("status-effect", () => {
  beforeAll(() => {
    i18next.init();
  });

  describe("NONE", () => {
    const statusEffect = StatusEffect.NONE;

    it("should return the obtain text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectObtainText(statusEffect, "");
      expect(text).toBe("statusEffect:none.obtain");
    });

    it("should return the source-obtain text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectObtainText(statusEffect, "PKM", "SOURCE");
      expect(text).toBe("statusEffect:none.obtainSource");
    });

    it("should return the activation text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectActivationText(statusEffect, "PKM");
      expect(text).toBe("statusEffect:none.activation");
    });

    it("should return the overlap text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectOverlapText(statusEffect, "PKM");
      expect(text).toBe("statusEffect:none.overlap");
    });

    it("should return the heal text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectHealText(statusEffect, "PKM");
      expect(text).toBe("statusEffect:none.heal");
    });

    it("should return the descriptor", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectDescriptor(statusEffect);
      expect(text).toBe("statusEffect:none.description");
    });
  });

  describe("POISON", () => {
    const statusEffect = StatusEffect.POISON;

    it("should return the obtain text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectObtainText(statusEffect, "");
      expect(text).toBe("statusEffect:poison.obtain");
    });

    it("should return the activation text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectActivationText(statusEffect, "");
      expect(text).toBe("statusEffect:poison.activation");
    });

    it("should return the descriptor", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectDescriptor(statusEffect);
      expect(text).toBe("statusEffect:poison.description");
    });

    it("should return the heal text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectHealText(statusEffect, "");
      expect(text).toBe("statusEffect:poison.heal");
    });

    it("should return the overlap text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectOverlapText(statusEffect, "");
      expect(text).toBe("statusEffect:poison.overlap");
    });
  });

  describe("TOXIC", () => {
    const statusEffect = StatusEffect.TOXIC;

    it("should return the obtain text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectObtainText(statusEffect, "");
      expect(text).toBe("statusEffect:toxic.obtain");
    });

    it("should return the activation text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectActivationText(statusEffect, "");
      expect(text).toBe("statusEffect:toxic.activation");
    });

    it("should return the descriptor", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectDescriptor(statusEffect);
      expect(text).toBe("statusEffect:toxic.description");
    });

    it("should return the heal text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectHealText(statusEffect, "");
      expect(text).toBe("statusEffect:toxic.heal");
    });

    it("should return the overlap text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectOverlapText(statusEffect, "");
      expect(text).toBe("statusEffect:toxic.overlap");
    });
  });

  describe("PARALYSIS", () => {
    const statusEffect = StatusEffect.PARALYSIS;

    it("should return the obtain text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectObtainText(statusEffect, "");
      expect(text).toBe("statusEffect:paralysis.obtain");
    });

    it("should return the activation text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectActivationText(statusEffect, "");
      expect(text).toBe("statusEffect:paralysis.activation");
    });

    it("should return the descriptor", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectDescriptor(statusEffect);
      expect(text).toBe("statusEffect:paralysis.description");
    });

    it("should return the heal text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectHealText(statusEffect, "");
      expect(text).toBe("statusEffect:paralysis.heal");
    });

    it("should return the overlap text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectOverlapText(statusEffect, "");
      expect(text).toBe("statusEffect:paralysis.overlap");
    });
  });

  describe("SLEEP", () => {
    const statusEffect = StatusEffect.SLEEP;

    it("should return the obtain text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectObtainText(statusEffect, "");
      expect(text).toBe("statusEffect:sleep.obtain");
    });

    it("should return the activation text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectActivationText(statusEffect, "");
      expect(text).toBe("statusEffect:sleep.activation");
    });

    it("should return the descriptor", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectDescriptor(statusEffect);
      expect(text).toBe("statusEffect:sleep.description");
    });

    it("should return the heal text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectHealText(statusEffect, "");
      expect(text).toBe("statusEffect:sleep.heal");
    });

    it("should return the overlap text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectOverlapText(statusEffect, "");
      expect(text).toBe("statusEffect:sleep.overlap");
    });
  });

  describe("FREEZE", () => {
    const statusEffect = StatusEffect.FREEZE;

    it("should return the obtain text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectObtainText(statusEffect, "");
      expect(text).toBe("statusEffect:freeze.obtain");
    });

    it("should return the activation text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectActivationText(statusEffect, "");
      expect(text).toBe("statusEffect:freeze.activation");
    });

    it("should return the descriptor", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectDescriptor(statusEffect);
      expect(text).toBe("statusEffect:freeze.description");
    });

    it("should return the heal text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectHealText(statusEffect, "");
      expect(text).toBe("statusEffect:freeze.heal");
    });

    it("should return the overlap text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectOverlapText(statusEffect, "");
      expect(text).toBe("statusEffect:freeze.overlap");
    });
  });

  describe("BURN", () => {
    const statusEffect = StatusEffect.BURN;

    it("should return the obtain text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectObtainText(statusEffect, "");
      expect(text).toBe("statusEffect:burn.obtain");
    });

    it("should return the activation text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectActivationText(statusEffect, "");
      expect(text).toBe("statusEffect:burn.activation");
    });

    it("should return the descriptor", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectDescriptor(statusEffect);
      expect(text).toBe("statusEffect:burn.description");
    });

    it("should return the heal text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectHealText(statusEffect, "");
      expect(text).toBe("statusEffect:burn.heal");
    });

    it("should return the overlap text", () => {
      vi.spyOn(i18next, "t").mockImplementation(tMock);
      const text = getStatusEffectOverlapText(statusEffect, "");
      expect(text).toBe("statusEffect:burn.overlap");
    });

    it("should increment turn count", () => {
      const status = new Status(StatusEffect.BURN);
      status.incrementTurn();
      expect(status.turnCount).toBe(1);
      status.incrementTurn();
      expect(status.turnCount).toBe(2);
    });

    it("should be a post turn status effect", () => {
      const status = new Status(StatusEffect.BURN);
      expect(status.isPostTurn()).toBe(true);
    });
  });

  afterEach(() => {
    vi.resetAllMocks();
  });
});
```
</details>

_It's not perfect because of the way i18n is set up in our tests (this will be fixed at a later point) but it gets the job done._

> _Click on the file-name to view the code_

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
-    activation: "{{pokemonNameWithAffix}} is fast asleep.",
-    overlap: "{{pokemonNameWithAffix}} is\nalready asleep!",
-    heal: "{{pokemonNameWithAffix}} woke up!"
+    name: "Dormido",
+    description: "dormido",
```
```suggestion
    name: "Dormir",
    description: "dormir",
```

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
-    overlap: "{{pokemonNameWithAffix}} is\nalready burned!",
-    heal: "{{pokemonNameWithAffix}} was\nhealed of its burn!"
+    name: "Quemado",
+    description: "quemado",
```
```suggestion
    description: "quemadura",
```

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
+    description: "envenenado ",
+    obtain: "¡{{pokemonNameWithAffix}}\nha sido envenenado!",
+    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido envenenado por {{sourceText}}!",
+    activation: "¡{{pokemonNameWithAffix}} se hizo daño\npor el veneno!",
```
```suggestion
    activation: "¡El veneno resta PS a {{pokemonNameWithAffix}}!",
```

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
+    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido envenenado por {{sourceText}}!",
+    activation: "¡{{pokemonNameWithAffix}} se hizo daño\npor el veneno!",
+    overlap: "¡{{pokemonNameWithAffix}} ya\nestá envenenado!",
+    heal: "¡{{pokemonNameWithAffix}} se\ncuró del envenenamiento!"
```
```suggestion
    heal: "¡{{pokemonNameWithAffix}} ya no\nestá envenenado!"
```

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
+    obtainSource: "¡{{pokemonNameWithAffix}} sufre parálisis por {{sourceText}}!\nQuizás no se pueda mover.",
+    activation: "¡{{pokemonNameWithAffix}} está paralizado!\n¡No se puede mover!",
+    overlap: "¡{{pokemonNameWithAffix}} ya\nestá paralizado!",
+    heal: "¡{{pokemonNameWithAffix}} se\ncuró de la parálisis !"
```
```suggestion
    heal: "¡{{pokemonNameWithAffix}} ya no\nestá paralizado!"
```

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
-    overlap: "{{pokemonNameWithAffix}} is\nalready poisoned!",
-    heal: "{{pokemonNameWithAffix}} was\ncured of its poison!"
+    name: "Envenenado",
+    description: "envenenado ",
```
```suggestion
    description: "envenenamiento",
```

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
-    overlap: "{{pokemonNameWithAffix}} is\nalready frozen!",
-    heal: "{{pokemonNameWithAffix}} was\ndefrosted!"
+    name: "Congelado",
+    description: "congelado",
```
```suggestion
    description: "congelamiento",
```

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
+    obtainSource: "¡{{pokemonNameWithAffix}}\nse ha quemado por {{sourceText}}!",
+    activation: "¡{{pokemonNameWithAffix}} se hizo daño\npor el quemado!",
+    overlap: "¡{{pokemonNameWithAffix}} ya\nestá quemado!",
+    heal: "¡{{pokemonNameWithAffix}} se\ncuró del quemado!"
```
```suggestion
    heal: "¡{{pokemonNameWithAffix}} ya no\nestá quemado!"
```

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
-    heal: "{{pokemonNameWithAffix}} woke up!"
+    name: "Dormido",
+    description: "dormido",
+    obtain: "¡{{pokemonNameWithAffix}}\nse durmió!",
+    obtainSource: "¡{{pokemonNameWithAffix}}\nse durmió por {{sourceText}}!",
```
```suggestion
    obtain: "¡{{pokemonNameWithAffix}}\nse ha dormido!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nse ha dormido\npor culpa de {{sourceText}}!",
```

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
+    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido congelado por {{sourceText}}!",
+    activation: "¡{{pokemonNameWithAffix}} está\ncongelado!",
+    overlap: "¡{{pokemonNameWithAffix}} ya\nestá congelado!",
+    heal: "¡{{pokemonNameWithAffix}} se\ndescongeló!"
```
```suggestion
    heal: "¡{{pokemonNameWithAffix}} se\nha descongelado!"
```

EnochG1 commented:
@flx-sta I commited following your suggestions,
except this code because my test code is in lokalisation(=localization) folder and I think feature test code and localization test code should be separate.
```
    it("should increment turn count", () => {
      const status = new Status(StatusEffect.BURN);
      status.incrementTurn();
      expect(status.turnCount).toBe(1);
      status.incrementTurn();
      expect(status.turnCount).toBe(2);
    });

    it("should be a post turn status effect", () => {
      const status = new Status(StatusEffect.BURN);
      expect(status.isPostTurn()).toBe(true);
    });
```

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -11,57 +11,57 @@ export const statusEffect: StatusEffectTranslationEntries = {
+    description: "quemado",
+    obtain: "¡{{pokemonNameWithAffix}}\nse ha quemado!",
+    obtainSource: "¡{{pokemonNameWithAffix}}\nse ha quemado por {{sourceText}}!",
+    activation: "¡{{pokemonNameWithAffix}} se hizo daño\npor el quemado!",
```
```suggestion
    activation: "¡{{pokemonNameWithAffix}} se resiente\nde las quemaduras!",
```

InnocentGameDev reviewed (commented):
es localization wasn't following Poké Corpus. Example:

https://abcboy101.github.io/poke-corpus/#query=hurt+by+its+burn&regex=false&caseInsensitive=true&common=true&script=true&collections=Red%2CGreen%2CBlue%2CYellow%2CGold%2CSilver%2CCrystal%2CRuby%2CSapphire%2CFireRed%2CLeafGreen%2CEmerald%2CDiamondPearl%2CPlatinum%2CHeartGoldSoulSilver%2CBlackWhite%2CBlack2White2%2CXY%2COmegaRubyAlphaSapphire%2CSunMoon%2CUltraSunUltraMoon%2CLetsGoPikachuLetsGoEevee%2CSwordShield%2CBrilliantDiamondShiningPearl%2CLegendsArceus%2CScarletViolet%2CColosseum%2CXD%2CBattleRevolution%2CBox%2CRanch%2CBank%2CHOME&languages=en-US%2Cen-GB

flx-sta commented:
> @flx-sta I commited following your suggestions, except this code because my test code is in lokalisation(=localization) folder and I think feature test code and localization test code should be separate.
> 
> ```
>     it("should increment turn count", () => {
>       const status = new Status(StatusEffect.BURN);
>       status.incrementTurn();
>       expect(status.turnCount).toBe(1);
>       status.incrementTurn();
>       expect(status.turnCount).toBe(2);
>     });
> 
>     it("should be a post turn status effect", () => {
>       const status = new Status(StatusEffect.BURN);
>       expect(status.isPostTurn()).toBe(true);
>     });
> ```

Yeah I didn't even write those :D That was the AI

flx-sta commented:
@EnochG1 you could add 1 more test to cover what happens if you pass an empty string as `sourceText`

EnochG1 commented:
@flx-sta I added it.

InnocentGameDev commented on src/locales/es/status-effect.ts:
```diff
@@ -0,0 +1,67 @@
+    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido gravemente envenenado por {{sourceText}}!",
+    activation: "¡{{pokemonNameWithAffix}} se hizo daño\npor el veneno!",
+    overlap: "¡{{pokemonNameWithAffix}} ya\nestá envenenado!",
+    heal: "¡{{pokemonNameWithAffix}} se\ncuró del envenenamiento!"
+  },
+  paralysis: {
+    name: "Parálisis",
+    description: "paralizado",
+    obtain: "¡{{pokemonNameWithAffix}} sufre parálisis!\nQuizás no se pueda mover.",
+    obtainSource: "¡{{pokemonNameWithAffix}} sufre parálisis por {{sourceText}}!\nQuizás no se pueda mover.",
+    activation: "¡{{pokemonNameWithAffix}} está paralizado!\n¡No se puede mover!",
+    overlap: "¡{{pokemonNameWithAffix}} ya\nestá paralizado!",
+    heal: "¡{{pokemonNameWithAffix}} ya no\nestá paralizado!"
+  },
+  sleep: {
+    name: "Dormir",
+    description: "dormir",
+    obtain: "¡{{pokemonNameWithAffix}}\nse ha dormido!",
+    obtainSource: "¡{{pokemonNameWithAffix}}\nse ha dormido\npor culpa de {{sourceText}}!",
+    activation: "¡{{pokemonNameWithAffix}} está/ndormido como un tronco.",
+    overlap: "¡{{pokemonNameWithAffix}} ya\nestá dormido!",
+    heal: "¡{{pokemonNameWithAffix}} se despertó!"
+  },
+  freeze: {
+    name: "Congelamiento",
+    description: "congelamiento",
+    obtain: "¡{{pokemonNameWithAffix}}\nha sido congelado!",
+    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido congelado por {{sourceText}}!",
+    activation: "¡{{pokemonNameWithAffix}} está\ncongelado!",
+    overlap: "¡{{pokemonNameWithAffix}} ya\nestá congelado!",
+    heal: "¡{{pokemonNameWithAffix}} se\nha descongelado!"
+  },
+  burn: {
+    name: "Quemadura",
+    description: "quemadura",
+    obtain: "¡{{pokemonNameWithAffix}}\nse ha quemado!",
+    obtainSource: "¡{{pokemonNameWithAffix}}\nse ha quemado por {{sourceText}}!",
+    activation: "¡{{pokemonNameWithAffix}} se resiente\nde las quemaduras!",
+    overlap: "¡{{pokemonNameWithAffix}} ya\nestá quemado!",
+    heal: "¡{{pokemonNameWithAffix}} ya no\nestá quemado!"
```
```suggestion
  none: {
    name: "Ninguno",
    description: "",
    obtain: "",
    obtainSource: "",
    activation: "",
    overlap: "",
    heal: ""
  },
  poison: {
    name: "Envenenamiento",
    description: "envenenamiento",
    obtain: "¡{{pokemonNameWithAffix}}\nha sido envenenado!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido envenenado por {{sourceText}}!",
    activation: "¡El veneno resta PS a {{pokemonNameWithAffix}}!",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá envenenado!",
    heal: "¡{{pokemonNameWithAffix}} ya no\nestá envenenado!"
  },
  toxic: {
    name: "Envenenamiento grave",
    description: "envenenamiento grave",
    obtain: "¡{{pokemonNameWithAffix}}\nha sido gravemente envenenado!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido gravemente envenenado por {{sourceText}}!",
    activation: "¡El veneno resta PS a {{pokemonNameWithAffix}}!",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá envenenado!",
    heal: "¡{{pokemonNameWithAffix}} ya no\nestá envenenado!"
  },
  paralysis: {
    name: "Parálisis",
    description: "parálisis",
    obtain: "¡{{pokemonNameWithAffix}} sufre parálisis!\nQuizás no se pueda mover.",
    obtainSource: "¡{{pokemonNameWithAffix}} sufre parálisis por {{sourceText}}!\nQuizás no se pueda mover.",
    activation: "¡{{pokemonNameWithAffix}} está paralizado!\n¡No se puede mover!",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá paralizado!",
    heal: "¡{{pokemonNameWithAffix}} ya no\nestá paralizado!"
  },
  sleep: {
    name: "Dormir",
    description: "dormir",
    obtain: "¡{{pokemonNameWithAffix}}\nse ha dormido!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nse ha dormido\npor culpa de {{sourceText}}!",
    activation: "¡{{pokemonNameWithAffix}} está/ndormido como un tronco.",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá dormido!",
    heal: "¡{{pokemonNameWithAffix}} se despertó!"
  },
  freeze: {
    name: "Congelamiento",
    description: "congelamiento",
    obtain: "¡{{pokemonNameWithAffix}}\nha sido congelado!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nha sido congelado por {{sourceText}}!",
    activation: "¡{{pokemonNameWithAffix}} está\ncongelado!",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá congelado!",
    heal: "¡{{pokemonNameWithAffix}} se\nha descongelado!"
  },
  burn: {
    name: "Quemadura",
    description: "quemadura",
    obtain: "¡{{pokemonNameWithAffix}}\nse ha quemado!",
    obtainSource: "¡{{pokemonNameWithAffix}}\nse ha quemado por {{sourceText}}!",
    activation: "¡{{pokemonNameWithAffix}} se resiente\nde las quemaduras!",
    overlap: "¡{{pokemonNameWithAffix}} ya\nestá quemado!",
    heal: "¡{{pokemonNameWithAffix}} ya no\nestá quemado!"
```

InnocentGameDev reviewed (commented):
Another check just to make sure

InnocentGameDev reviewed (approved):
ES translation seems good now

GoldTra commented:
> ES translation seems good now
There is a problem with the PokémonNameWithAffix because it always includes "El" in the phrase, and it will appear in the game as "El veneno resta PS a el abra salvaje."

EnochG1 commented:
@GoldTra,
@InnocentGameDev discussed it in discord (wiki discord, with translation team)
And it's not this PR's issue (but spanish pokemonNameWithAffix's)
So I would request to merge this PR.
Maybe we have to solve it in next task.

Adri1 commented on src/locales/fr/status-effect.ts:
```diff
@@ -0,0 +1,67 @@
+  },
+  toxic: {
+    name: "Empoisonnement grave",
+    description: "empoisonnement grave",
+    obtain: "{{pokemonNameWithAffix}} est\ngravement empoisonné !",
+    obtainSource: "{{pokemonNameWithAffix}} est\ngravement empoisonné par {{sourceText}} !",
+    activation: "{{pokemonNameWithAffix}}\nsouffre du poison !",
+    overlap: "{{pokemonNameWithAffix}} est\ndéjà empoisonné.",
+    heal: "{{pokemonNameWithAffix}} n’est\nplus empoisonné !"
+  },
+  paralysis: {
+    name: "Paralysie",
+    description: "paralysie",
+    obtain: "{{pokemonNameWithAffix}} est paralysé !\nIl aura du mal à attaquer !",
+    obtainSource: "{{pokemonNameWithAffix}} est paralysé\npar {{sourceText}} ! Il aura du mal à attaquer !",
+    activation: "{{pokemonNameWithAffix}} est paralysé !\nIl n’a pas pu attaquer !",
+    overlap: "{{pokemonNameWithAffix}} est\ndéjà paralysé.",
+    heal: "{{pokemonNameWithAffix}} n’est\nplus paralysé !"
+  },
+  sleep: {
+    name: "Sommeil",
+    description: "sommeil",
+    obtain: "{{pokemonNameWithAffix}}\ns’est endormi !",
+    obtainSource: "{{pokemonNameWithAffix}} est\nendormi par {{sourceText}} !",
+    activation: "{{pokemonNameWithAffix}}\ndort profondément.",
+    overlap: "{{pokemonNameWithAffix}}\ndort déjà.",
+    heal: "{{pokemonNameWithAffix}}\nse réveille !"
+  },
+  freeze: {
+    name: "Gelé",
+    description: "gelé",
+    obtain: "{{pokemonNameWithAffix}} est\ngelé !",
+    obtainSource: "{{pokemonNameWithAffix}} est\ngelé par {{sourceText}} !",
+    activation: "{{pokemonNameWithAffix}}est gelé !\nIl ne peut plus attaquer !",
+    overlap: "{{pokemonNameWithAffix}} est\ndéjà gelé.",
+    heal: "{{pokemonNameWithAffix}} n’est\nplus gelé !"
+  },
+  burn: {
+    name: "Brulure",
+    description: "brulure",
```
```suggestion
    description: "empoisonné",
    obtain: "{{pokemonNameWithAffix}} est\nempoisonné !",
    obtainSource: "{{pokemonNameWithAffix}} est\nempoisonné par {{sourceText}} !",
    activation: "{{pokemonNameWithAffix}}\nsouffre du poison !",
    overlap: "{{pokemonNameWithAffix}} est\ndéjà empoisonné.",
    heal: "{{pokemonNameWithAffix}} n’est\nplus empoisonné !"
  },
  toxic: {
    name: "Empoisonnement grave",
    description: "gravement empoisonné",
    obtain: "{{pokemonNameWithAffix}} est\ngravement empoisonné !",
    obtainSource: "{{pokemonNameWithAffix}} est\ngravement empoisonné par {{sourceText}} !",
    activation: "{{pokemonNameWithAffix}}\nsouffre du poison !",
    overlap: "{{pokemonNameWithAffix}} est\ndéjà empoisonné.",
    heal: "{{pokemonNameWithAffix}} n’est\nplus empoisonné !"
  },
  paralysis: {
    name: "Paralysie",
    description: "paralysé",
    obtain: "{{pokemonNameWithAffix}} est paralysé !\nIl aura du mal à attaquer !",
    obtainSource: "{{pokemonNameWithAffix}} est paralysé\npar {{sourceText}} ! Il aura du mal à attaquer !",
    activation: "{{pokemonNameWithAffix}} est paralysé !\nIl n’a pas pu attaquer !",
    overlap: "{{pokemonNameWithAffix}} est\ndéjà paralysé.",
    heal: "{{pokemonNameWithAffix}} n’est\nplus paralysé !"
  },
  sleep: {
    name: "Sommeil",
    description: "endormi",
    obtain: "{{pokemonNameWithAffix}}\ns’est endormi !",
    obtainSource: "{{pokemonNameWithAffix}} est\nendormi par {{sourceText}} !",
    activation: "{{pokemonNameWithAffix}}\ndort profondément.",
    overlap: "{{pokemonNameWithAffix}}\ndort déjà.",
    heal: "{{pokemonNameWithAffix}}\nse réveille !"
  },
  freeze: {
    name: "Gelé",
    description: "gelé",
    obtain: "{{pokemonNameWithAffix}} est\ngelé !",
    obtainSource: "{{pokemonNameWithAffix}} est\ngelé par {{sourceText}} !",
    activation: "{{pokemonNameWithAffix}}est gelé !\nIl ne peut plus attaquer !",
    overlap: "{{pokemonNameWithAffix}} est\ndéjà gelé.",
    heal: "{{pokemonNameWithAffix}} n’est\nplus gelé !"
  },
  burn: {
    name: "Brulure",
    description: "brulé",
```

Adri1 reviewed (commented):
French adjustments
